lib/listener.js

```diff
@@ -0,0 +1,68 @@
+'use strict';
+
+const co = require('co');
+const EventEmitter = require('events');
+const bus = require('./client');
+
+/**
+ * Return a bus listener with helper methods to register listeners and listen to the bus' messages
+ * The instance inherits EventEmitter and can emit following events :
+ * - connected : when the listener is connected to the bus, takes no arguments
+ *
+ * @return {Object} bus listener instance
+ */
+function* createBusListener(url) {
+  const _queues = [];
+  const _handlers = {};
+
+  const instance = Object.create(EventEmitter.prototype);
+  instance.use = use;
+  instance.listen = listen;
+  instance.connection = null;
+  instance.client = null;
+  return instance;
+
+  /**
+   * Register a new handler for a given queue and key
+   * @param {String} queue Queue
+   * @param {String} key Key
+   * @param {Function} handler Handler
+   */
+  function use(queue, key, handler) {
+    if (!_handlers[queue]) {
+      _queues.push(queue);
+      _handlers[queue] = {};
+    }
+    _handlers[queue][key] = handler;
+  }
+
+  /**
+   * Start listening on registered handlers. Handlers that are registered later will not be listened
+   * to
+   *
+   * @param {String} exchange Exchange name
+   */
+  function* listen(exchange) {
+    if (instance.connection) return instance.connection;
```

Comment:
Calling a reference to `instance` here is very flimsy... Are you sure that it will be the right one in every instance of your bus client ? Are you sure it will event work ? I think you'd get an `undefined` error. Since you have not "bound" `listen` to your instance (just "linked"), then I don't see why this should work.